lib/api/build.rs

```diff
@@ -1,178 +1,215 @@
-        // -L native=/path/runng/target/debug/build/runng-sys-abc1234/out
-        // That contains output from cmake
-        println!(
-            "cargo:rustc-link-search=native={}",
-            dst.join("build").display()
+    if objcopy.is_none() {
+        panic!(
+            "No program akin to `objcopy` found\nI searched for these programs in your path: {}",
+            objcopy_names.join(", ")
         );
-        println!("cargo:rustc-link-lib=vmlib");
-
-        let bindings = bindgen::Builder::default()
-            .header(
-                wamr_dir
-                    .join("core/iwasm/include/wasm_c_api.h")
-                    .to_str()
-                    .unwrap(),
-            )
-            .header(
-                wamr_dir
-                    .join("core/iwasm/include/wasm_export.h")
-                    .to_str()
-                    .unwrap(),
-            )
-            .derive_default(true)
-            .derive_debug(true)
-            .generate()
-            .expect("Unable to generate bindings");
-        let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
-        bindings
-            .write_to_file(out_path.join("bindings.rs"))
-            .expect("Couldn't write bindings");
     }
 
-    #[cfg(feature = "wasmi")]
-    {
-        use std::{env, path::PathBuf};
-
-        let bindings = bindgen::Builder::default()
-            .header(
-                PathBuf::from(std::env::var("DEP_WASMI_C_API_INCLUDE").unwrap())
-                    .join("wasm.h")
-                    .to_string_lossy(),
-            )
-            .derive_default(true)
-            .derive_debug(true)
-            .generate()
-            .expect("Unable to generate bindings for `wasmi`!");
-        let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
-        bindings
-            .write_to_file(out_path.join("bindings.rs"))
-            .expect("Couldn't write bindings");
+    let objcopy = objcopy.unwrap();
+
+    unsafe {
+        let syms: Vec<String> = WAMR_RENAMED
+            .iter()
+            .map(|(old, new)| format!("--redefine-sym=_{old}={new}"))
+            .collect();
+        dbg!(std::process::Command::new(objcopy)
+            .args(syms)
+            .arg(dst.join("build").join("libvmlib.a").display().to_string())
+            .arg(dst.join("build").join("libwamr.a").display().to_string()))
+        .output()
+        .unwrap();
     }
 
-    #[cfg(feature = "v8")]
-    {
-        use std::{env, path::PathBuf};
+    println!(
+        "cargo:rustc-link-search=native={}",
+        dst.join("build").display()
+    );
+    println!("cargo:rustc-link-lib=wamr");
+}
 
-        let url = match (
+#[cfg(feature = "v8")]
+fn build_v8() {
+    use bindgen::callbacks::ParseCallbacks;
+    use std::{env, path::PathBuf};
+
+    let header_url = "https://raw.githubusercontent.com/laper32/v8-cmake/refs/heads/master/v8/third_party/wasm-api/wasm.h";
```

Comment:
Build steps REALLY should not download random stuff at build time... can this be embedded in the repo?

I guess v8 is pretty huge?

Reply:
Yep, you're totally right. I forgot to embed that file in the repo.

Reply:
[f381470](https://github.com/wasmerio/wasmer/pull/5243/commits/f3814701f8abb16d6481d2b91816583982b6220c)